src/nvim/buffer.c

```diff
@@ -4590,9 +4590,8 @@ wipe_buffer (
   if (buf->b_fnum == top_file_num - 1)
     --top_file_num;
 
-  if (!aucmd)               /* Don't trigger BufDelete autocommands here. */
-    block_autocmds();
-  close_buffer(NULL, buf, DOBUF_WIPE, FALSE);
-  if (!aucmd)
-    unblock_autocmds();
+  // Don't trigger BufDelete autocommands here.
+  WITH_MAYBE_BLOCK_AUTOCMDS(!aucmd, {
+    close_buffer(NULL, buf, DOBUF_WIPE, FALSE);
+  });
```

Comment:
This is the only place this version of the macro got used. Looking at it further, I wonder if it would be better just to block the autocmds in the few places that call this method and want them blocked. And then remove the `aucmd` variable as well.